types/artifacts.d.ts

```diff
@@ -1020,6 +1020,23 @@ export interface TraceEvent {
   };
 }
 
+declare module Trace {
+  /**
+   * Base event of a `ph: 'X'` 'complete' event. Extend with `name` and `args` as
+   * needed.
+   */
+  interface CompleteEvent {
```

Comment:
[kind of harkens back to your trace types, @connorjclark :)

I kind of like this pattern of building up the complete event compared to the `Omit<>`+build pattern used here: 
https://github.com/GoogleChrome/lighthouse/blob/0fcf4a78aed37bad1aa71691c74154ec23c91444/lighthouse-core/lib/tracehouse/trace-processor.js#L21-L23